test-crawler/extractor/extractor.go

```diff
@@ -78,35 +97,48 @@ func getFileContent(filePath string) (content string, err error) {
 	return string(src), nil
 }
 
-func parseContent(content string, treeCursor *sitter.TreeCursor, filePath string) ([]c.Scenario, *Metadata, error) {
-	var scenarios []c.Scenario
-	var metadata *Metadata
+func parseContent(content string, treeCursor *sitter.TreeCursor, filePath string) (*FileData, error) {
+	fileData := &FileData{}
 
 	var annotationParser a.Parser
 
-	metadata = getMetadata(content, treeCursor, &annotationParser)
+	fileData.Metadata = getMetadata(content, treeCursor, &annotationParser)
 	functions := getFunctionNodes(content, treeCursor, &annotationParser)
 
 	for _, function := range functions {
 
 		behaviors := findBehaviorsFromNode(content, function.Node)
+		var callExpressions []string = nil
+		if len(behaviors) > 0 {
+			callExpressions = findCallExprFromNode(content, function.Node)
+		}
 
-		scenarios = append(scenarios, makeCollectorScenario(filePath, function.Name, behaviors))
+		fileData.Functions = append(fileData.Functions, makeCollectorScenario(filePath, function.Name, behaviors, callExpressions))
 
 	}
 
-	return scenarios, metadata, nil
+	return fileData, nil
+}
+
+// if non of the files contain any behaviors, the file should be ignored
```

Comment:
I assume you meant `if none of the functions contain any behaviors`?

Reply:
You are right, fixing..